src/node.rs

```diff
@@ -301,33 +331,409 @@ impl Taffy {
     }
 }
 
-/// Internal node id.
-pub(crate) type NodeId = usize;
+#[test]
+/// New should be an empty with allocated size of 16
+fn test_new_is_default() {
+    // FIXME: This does not work unless we implement `PartialEq`, `Eq` on most types in the
+    // Is this a useful test to keep?
+    // assert_eq!(Taffy::new(), Taffy::default());
+}
 
-/// The identifier of a [`Node`]
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
-#[cfg_attr(not(any(feature = "std", feature = "alloc")), derive(hash32_derive::Hash32))]
-pub(crate) struct Id(usize);
+// #[test]
+// fn new_should_allocate_default_capacity() {
+//     const DEFAULT_CAPACITY: usize = 16;
+//     let taffy = Taffy::new();
+
+//     assert!(taffy.nodes_to_ids.capacity() >= DEFAULT_CAPACITY);
+//     assert!(taffy.ids_to_nodes.capacity() >= DEFAULT_CAPACITY);
+//     assert!(taffy.forest.children.capacity() >= DEFAULT_CAPACITY);
+//     assert!(taffy.forest.parents.capacity() >= DEFAULT_CAPACITY);
+//     assert!(taffy.forest.nodes.capacity() >= DEFAULT_CAPACITY);
+// }
+
+// #[test]
+// fn test_with_capacity() {
+//     const CAPACITY: usize = 8;
+//     let taffy = Taffy::with_capacity(CAPACITY);
+
+//     assert!(taffy.nodes_to_ids.capacity() >= CAPACITY);
+//     assert!(taffy.ids_to_nodes.capacity() >= CAPACITY);
+//     assert!(taffy.forest.children.capacity() >= CAPACITY);
+//     assert!(taffy.forest.parents.capacity() >= CAPACITY);
+//     assert!(taffy.forest.nodes.capacity() >= CAPACITY);
+// }
+
+// #[test]
+// fn allocate_node() {
+//     let mut taffy = Taffy::new();
+//     // let old_last_id = taffy.allocator.last_id.;
+//     let id = taffy.allocate_node();
+
+//     todo!("How to verify that this is working?");
+// }
```

Comment:
These I don't know if are useful or not. They fail on `[Features:None]` as these are private methods